main.nf

```diff
@@ -1,55 +1,169 @@
+    Validate parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+include { validateParameters; } from 'plugin/nf-validation'
+validateParameters()
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Import modules/subworkflows
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+include { FASTQC } from './modules/nf-core/fastqc/main'
+include { MULTIQC } from './modules/nf-core/multiqc/main'
+include { OUTRIDER } from './NextflowModules/Outrider/1.20.0/main'
+include { SAMTOOLS_INDEX } from './modules/nf-core/samtools/index/main'
+include { SAMTOOLS_MERGE } from './modules/nf-core/samtools/merge/main'
+include { STAR_ALIGN } from './modules/nf-core/star/align/main'
+include { SUBREAD_FEATURECOUNTS } from './modules/nf-core/subread/featurecounts/main'
+include { TRIMGALORE } from './modules/nf-core/trimgalore/main'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Main workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
 
+workflow {
+    ch_fastq = Channel.fromFilePairs("$params.input/*_R{1,2}_001.fastq.gz")
+        .map {
+            meta, fastq ->
+            def fmeta = [:]
+            // Set meta.id
+            fmeta.id = meta
+	    // Set meta.single_end
+            if (fastq.size() == 1) {
+                fmeta.single_end = true
+            } else {
+                fmeta.single_end = false
+            }
+            [ fmeta, fastq ]
+        }
+
+    fastq_to_bam(ch_fastq)
+    featurecounts(fastq_to_bam.out)
+    outrider(featurecounts.out)
+    QC(ch_fastq)
+}
 
-params.reads = '/hpc/diaggen/users/Behzad/bam/PMABM000_fastq/*_r{1,2}.fastq'
-params.transcripts = '/hpc/diaggen/users/Behzad/bam/STAR/hg38_genome/gencode.v39.transcripts.fa'
 
+workflow fastq_to_bam{
+    take: ch_fastq
+    main:
+    // Reference file channels
+    ch_star_index = Channel.fromPath(params.star_index).map {star_index -> [star_index.getSimpleName(), star_index] }
+    ch_gtf = Channel.fromPath(params.gtf).map { gtf -> [gtf.getSimpleName(), gtf] }
+    
+    TRIMGALORE(
+        ch_fastq
+    )
+    
+    STAR_ALIGN(
+        TRIMGALORE.out.reads,
+        ch_star_index.first(),
+        ch_gtf.first(),
+        false,
+        'illumina',
+        'UMCU Genetics'
+    )
+
+    SAMTOOLS_MERGE(
+        STAR_ALIGN.out.bam_sorted.map {
+            meta, bam ->
+                new_id = meta.id.split('_')[0]
+                [ meta + [id: new_id], bam ]
+        }.groupTuple(),
+        [ [ id:'null' ], []],
+        [ [ id:'null' ], []],
+    )
+
+    SAMTOOLS_INDEX ( SAMTOOLS_MERGE.out.bam )
+
+    SAMTOOLS_MERGE.out.bam
+        .join(SAMTOOLS_INDEX.out.bai)
+        .set { ch_bam_bai }
+
+    emit:
+	SAMTOOLS_MERGE.out.bam
+}
 
-process INDEX {
 
-    input:
-    path fasta
+workflow featurecounts{
```

Comment:
Nextflow uses lower_case and underscore as separator for (sub)workflows. Like workflow my_pipeline {.
So therefor I use featurecounts, although it could be feature_counts.I added the space

Reply:
Good to know, thanks.
https://nf-co.re/docs/contributing/modules#naming-conventions 
https://nf-co.re/docs/contributing/subworkflows#naming-conventions